packages/lib/shim-init-node.test.ts

```diff
@@ -16,4 +17,15 @@ describe('shim-init-node', () => {
 
 		expect(resource.mime).toBe('application/pdf');
 	});
+
+	test('should keep file extension from filename if mime type is not recognized', async () => {
```

Comment:
I don't think that's what this feature is about?

The mime type **is** recognised, but we still want to keep the file extension.

Reply:
Same for the PR title, it's not quite right I think

Reply:
I wasn't sure how to explain it better, I'm open to suggestions.

The thing is that there are two places we try to assign the mime type to the resource, like I pointed out in the main body of the PR. 

- The first one is when we get the file extension and try to find the mime with `mimeUtils.fromFilename`
- If no mime is assigned we use `file-type` to try to find that information

Reply:
So I think the problem is that the mime type is actually known and detected, it's for example "application/zip" but the problem is that the file doesn't have the expected .zip file extension. But the result is that we keep the file extension if one is provided - in that case we ignore the mime type.

So maybe "it should preserve the file extension if one is provided regardless of the mime type"?

Reply:
I think that makes a lot of sense, thanks for the help.